source/uri/Endpoint.ooc

```diff
@@ -0,0 +1,53 @@
+Endpoint: class {
+	_host: VectorList<Text>
+	_port: Int
+	host: VectorList<Text> {
+		get {
+			result := VectorList<Text> new()
+			for (i in 0 .. this _host count) {
+				result add(this _host[i] take())
+			}
+			result
+		}
+	}
+	port: Int {
+		get { this _port }
+	}
+	init: func (=_host, =_port)
+	free: override func {
+		for (i in 0 .. this _host count)
+			this _host[i] free(Owner Receiver)
+		this _host free()
+		super()
+	}
+	toText: func -> Text {
+		result := Text new()
+		host := this host
+		for (i in 0 .. host count)
+			result += i == 0 ? host[i] : t"." + host[i]
+		if (this port != 0) {
+			portString := this port toString()
+			result += t":" + Text new(portString)
+			portString free()
+		}
+		host free()
+		result
+	}
+	parse: static func(data: Text) -> This {
+		result: This
+		d := data take()
+		if (!d isEmpty) {
+			splitted := d split(t":")
+			port := 0
+			if (splitted count == 2)
+				port = splitted remove() toInt()
+			host := splitted remove() split(t".")
```

Comment:
Ah ok, if d is not empty and does not contain `:` split will return the whole string